### PR TITLE
chore: Add option to generate frontmatter to gendocs.

### PR DESCRIPTION
**Pull request by rhuss**
This is needed for latter inclusion in knative.dev. Related to #639.

### CODE REVIEW

File: hack/generate-docs.go [hunk 1]
```diff
@@ -29,9 +33,45 @@ func main() {
...
+func addFrontMatter(fileName string) string {
+	// Convert to a title
+	title := filepath.Base(fileName)
+	title = title[0:len(title) - len(filepath.Ext(title))]
+	title = strings.ReplaceAll(title, "_", " ")
+	ret := `
+---
+title: "%s"
+#linkTitle: "OPTIONAL_ALTERNATE_NAV_TITLE"
+weight: 5
```
Comment (navidshaikh) [inline hack/generate-docs.go]: How does `weight` property work for docs?

Comment (navidshaikh) [reply hack/generate-docs.go]: nvm, for content ordering.

**Comment by navidshaikh** [comment]
Tested locally, works fine!

README change causing the build tests failure..

File: README.md [hunk 2]
```diff
@@ -1,42 +1,36 @@
-# Knative Client
+# Kn
 
-This section outlines best practices for the Knative developer experience. It is a reference for Knative CLI implementation, and a reference for Knative client libraries.
+The Knative client `kn` is your door to the [Knative](https://knative.dev) world.
```
Comment (maximilien) [inline README.md]: Why is this file changed in addition to the #677

**Comment by rhuss** [comment]
@maximilien  this PR is part of the doc automation described in #639 and is used to include the generated `kn` reference documentation into the `knative.dev` website. On our side we are just offering the way how create the documentation with frontmatter. The documentation build then will checkout the this repository, call the doc generation and commits the generated file into the website. This second part then has still to be done.

File: README.md (hunk 2 above)
Comment (rhuss) [reply README.md]: sorry, I mixed up to PRs. Should be fixed now.

**Comment by rhuss** [comment]
@maximilien any objection to lgtm this ? The change is non-intrusive, but would be the first step for reference manual integration to knative.dev
